apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/spring/webmvc/ViewRenderInstrumentation.java

```diff
@@ -0,0 +1,123 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.spring.webmvc;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.web.servlet.view.AbstractView;
+import org.thymeleaf.spring4.view.AbstractThymeleafView;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+public class ViewRenderInstrumentation extends ElasticApmInstrumentation {
+
+    private static final String SPAN_TYPE = "template";
+    private static final String SPAN_ACTION = "render";
+    private static final String DISPATCHER_SERVLET_RENDER_METHOD = "View#render";
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return ViewRenderAdviceService.class;
+    }
+
+    public static class ViewRenderAdviceService {
+
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        public static void beforeExecute(@Advice.Local("span") @Nullable Span span,
+                                         @Advice.This @Nullable Object thiz) {
+            if (tracer == null || tracer.getActive() == null) {
+                return;
+            }
+            final TraceContextHolder<?> parent = tracer.getActive();
+
+            String viewClassName = thiz.getClass().getSimpleName().replace("View", "");
```

Comment:
We can be a bit more efficient here. 

Similar to this (I have not tested it)

```java
StringBuilder name = span.getAndOverrideName(PRIO_DEFAULT);
if (name != null) {
    String className = thiz.getClass().getName();
    name.append(className, className.lastIndexOf('.'), className.lastIndexOf("View"));
}
```

Reply:
replaced on:
```
 String className = thiz.getClass().getName();
            span = parent.createSpan()
                .withType(SPAN_TYPE)
                .withSubtype(className.substring(className.lastIndexOf('.')+1, className.lastIndexOf("View")))
                .withAction(SPAN_ACTION)
                .withName(DISPATCHER_SERVLET_RENDER_METHOD);

```

Reply:
Substring still creates a new object. With my suggestion, there would be no allocation involved as it just copies over a range of characters from the class name string to the string builder.

Reply:
oh, right 🤦‍♂️
nevermind then